.jenkinsci/debug-build.groovy

```diff
@@ -52,63 +101,43 @@ def doDebugBuild(coverageEnabled=false) {
       + " -e IROHA_POSTGRES_USER=${env.IROHA_POSTGRES_USER}"
       + " -e IROHA_POSTGRES_PASSWORD=${env.IROHA_POSTGRES_PASSWORD}"
-      + " --network=${env.IROHA_NETWORK}"
-      + " -v /var/jenkins/ccache:${CCACHE_DIR}"
-      + " -v /tmp/${GIT_COMMIT}-${BUILD_NUMBER}:/tmp/${GIT_COMMIT}") {
-
-      def scmVars = checkout scm
-      def cmakeOptions = ""
-      if ( coverageEnabled ) {
-        cmakeOptions = " -DCOVERAGE=ON "
-      }
-      env.IROHA_VERSION = "0x${scmVars.GIT_COMMIT}"
-      env.IROHA_HOME = "/opt/iroha"
-      env.IROHA_BUILD = "${env.IROHA_HOME}/build"
-
-      sh """
-        ccache --version
-        ccache --show-stats
-        ccache --zero-stats
-        ccache --max-size=5G
-      """
-      sh """
-        cmake \
-          -DTESTING=ON \
-          -H. \
-          -Bbuild \
-          -DCMAKE_BUILD_TYPE=Debug \
-          -DIROHA_VERSION=${env.IROHA_VERSION} \
-          ${cmakeOptions}
-      """
-      sh "cmake --build build -- -j${parallelism}"
-      sh "ccache --show-stats"
-      if ( coverageEnabled ) {
-        sh "cmake --build build --target coverage.init.info"
-      }
-      def testExitCode = sh(script: """cd build && ctest --output-on-failure""", returnStatus: true)
-      if (testExitCode != 0) {
-        currentBuild.result = "UNSTABLE"
-      }
-      if ( coverageEnabled ) {
-        sh "cmake --build build --target cppcheck"
-        // Sonar
-        if (env.CHANGE_ID != null) {
-          sh """
-            sonar-scanner \
-              -Dsonar.github.disableInlineComments \
-              -Dsonar.github.repository='${DOCKER_REGISTRY_BASENAME}' \
-              -Dsonar.analysis.mode=preview \
-              -Dsonar.login=${SONAR_TOKEN} \
-              -Dsonar.projectVersion=${BUILD_TAG} \
-              -Dsonar.github.oauth=${SORABOT_TOKEN} \
-              -Dsonar.github.pullRequest=${CHANGE_ID}
-          """
+      + " --network=${env.IROHA_NETWORK}") {
+        def testExitCode = sh(script: """cd build && ctest --output-on-failure -R '${testList}' """, returnStatus: true)
+        if (testExitCode != 0) {
+          currentBuild.result = "UNSTABLE"
         }
-        sh "cmake --build build --target coverage.info"
-        sh "python /tmp/lcov_cobertura.py build/reports/coverage.info -o build/reports/coverage.xml"
-        cobertura autoUpdateHealth: false, autoUpdateStability: false, coberturaReportFile: '**/build/reports/coverage.xml', conditionalCoverageTargets: '75, 50, 0', failUnhealthy: false, failUnstable: false, lineCoverageTargets: '75, 50, 0', maxNumberOfBuilds: 50, methodCoverageTargets: '75, 50, 0', onlyStable: false, zoomCoverageChart: false
       }
     }
+}
+
+def doPostCoverageCoberturaStep() {
+  if (env.NODE_NAME.contains('x86_64')) {
+    sh "docker load -i ${JENKINS_DOCKER_IMAGE_DIR}/${DOCKER_IMAGE_FILE}"
+  }
+  def iC = docker.image("${DOCKER_AGENT_IMAGE}")
+  iC.inside() {
+    def step = load ".jenkinsci/mac-debug-build.groovy"
```

Comment:
Same for coverage, I don't think any other platform besides gcc/Linux is required.